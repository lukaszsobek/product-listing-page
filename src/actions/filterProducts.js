import { updatePage, toggleFilter } from "./";

const filterProducts = (filterType, filterValue) => {
    return dispatch => {
        dispatch(toggleFilter(filterType, filterValue));
        dispatch(updatePage());
    }
}

export default filterProducts;
