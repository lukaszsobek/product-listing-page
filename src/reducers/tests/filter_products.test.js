import filterProducts from "../filter_products";

describe("filterProducts reducer returns",() => {

    const products = [
        { colors: ["blue"], category: "one" },
        { colors: ["green"], category: "two" },
        { colors: ["blue"], category: "two" }
    ];

    const state = {
        activeFilters: { categories: [], colors: [] },
        products
    }

    it("all products if no filter set",() => {
        expect(filterProducts(state).filteredProducts).toMatchSnapshot();
    });

    it("only color products with color filter",() => {
        state.activeFilters.colors = ["blue"];
        expect(filterProducts(state).filteredProducts).toMatchSnapshot();
    });

    it("only categories with categories filter",() => {
        state.activeFilters.categories = ["one"];
        expect(filterProducts(state).filteredProducts).toMatchSnapshot();
    });

    it("only items with both filters set",() => {
        state.activeFilters.categories = ["two"];
        state.activeFilters.colors = ["green"];
        expect(filterProducts(state).filteredProducts).toMatchSnapshot();
    });
});
