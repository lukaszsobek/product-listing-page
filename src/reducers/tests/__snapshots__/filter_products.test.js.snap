// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterProducts reducer returns all products if no filter set 1`] = `
Array [
  Object {
    "category": "one",
    "colors": Array [
      "blue",
    ],
  },
  Object {
    "category": "two",
    "colors": Array [
      "green",
    ],
  },
  Object {
    "category": "two",
    "colors": Array [
      "blue",
    ],
  },
]
`;

exports[`filterProducts reducer returns only categories with categories filter 1`] = `
Array [
  Object {
    "category": "one",
    "colors": Array [
      "blue",
    ],
  },
]
`;

exports[`filterProducts reducer returns only color products with color filter 1`] = `
Array [
  Object {
    "category": "one",
    "colors": Array [
      "blue",
    ],
  },
  Object {
    "category": "two",
    "colors": Array [
      "blue",
    ],
  },
]
`;

exports[`filterProducts reducer returns only items with both filters set 1`] = `
Array [
  Object {
    "category": "two",
    "colors": Array [
      "green",
    ],
  },
]
`;
